src/core/processing/qgsprocessingutils.cpp

```diff
@@ -737,11 +738,13 @@ QgsFeatureSink *QgsProcessingUtils::createFeatureSink( QString &destination, Qgs
       // use QgsVectorFileWriter for OGR destinations instead of QgsVectorLayerImport, as that allows
       // us to use any OGR format which supports feature addition
       QString finalFileName;
+      QString finalLayerName;
       QgsVectorFileWriter::SaveVectorOptions saveOptions;
       saveOptions.fileEncoding = options.value( QStringLiteral( "fileEncoding" ) ).toString();
+      saveOptions.layerName = !layerName.isEmpty() ? layerName : options.value( QStringLiteral( "layerName" ) ).toString();
```

Comment:
Can you add a test to testqgsprocessing.cpp checking the behavior of the layerName handling?